libraries/AP_IQ_Motor/AP_IQ_Motor.cpp

```diff
@@ -0,0 +1,263 @@
+    _iq_uart = serial_manager->find_serial(AP_SerialManager::SerialProtocol_IQ, 0);
+
+    // check if the uart exists
+    if (_iq_uart) {
+        //Grab the number of IFCI values the user wants to use
+        const int8_t length = _broadcast_length.get();
+
+        //Limit the user's length parameter to only the number of available channels
+        _total_channels = NUM_SERVO_CHANNELS > length ? length : NUM_SERVO_CHANNELS;
+
+        //Find the position of the first and last motor that we will request telemetry from
+        const uint8_t bitmask = _telemetry_bitmask.get();
+        bool found_first = false;
+        for (uint8_t motor = 0; motor < NUM_SERVO_CHANNELS; ++motor) {
+            if (bitmask & 1 << motor) {
+                if (found_first) {
+                    _last_telem_motor = motor;
+                } else {
+                    _first_telem_motor = motor;
+                    found_first = true;
+                }
+            }
+        }
+    }
+}
+
+uint8_t AP_IQ_Motor::add_client(ClientAbstract *new_client)
+{
+    //Make sure that there is enough space to add a new client
+    if (_client_size < _client_limit) { // TODO check if this client type already exists and don't add if it does. Instead return the pointer?
+        _clients[_client_size] = new_client;
+        _client_size++;
+        return 1;
+    }
+
+    return 0;
+}
+
+uint8_t AP_IQ_Motor::find_client(ClientAbstract **set_client, uint8_t client_type, uint8_t obj_idn)
+{
+    //Go and try to find the client specified in the function call
+    for (int client = 0; client < _client_size; ++client) {
+        if (_clients[client]->obj_idn_ == obj_idn && _clients[client]->type_idn_ == client_type) {
+            *set_client = _clients[client];
+            return 1;
+        }
+    }
+    return 0;
+}
+
+void AP_IQ_Motor::update(void)
+{
+    //Only initialize once
+    if (!_initialized) {
+        _initialized = true; // check if init worked
+        init();
+    }
+
+    //If we are ever unable to find an instance of our communication protocol, stop trying to update
+    if (_iq_uart == nullptr) {
+        return;
+    }
+
+    // TODO: generate general broadcast output message to esc prop from channels array
+    // TODO: send telemetry request for current telemetry motor
+    read();
+    update_motor_outputs();
+    if (_telemetry_bitmask.get() != 0) {
+        update_telemetry();
+    }
+    write();
+
+}
+
+void AP_IQ_Motor::update_motor_outputs()
+{
+    if (hal.util->get_soft_armed()) {
+        //Go through all of our outputs, and, if possible, update its value
+        uint16_t raw_control_values[_total_channels];
+        for (unsigned ii = 0; ii < _total_channels; ++ii) {
+            SRV_Channel *c = SRV_Channels::srv_channel(ii);
+            if (c == nullptr) {
+                continue;
+            }
+
+            uint16_t output = uint16_t(constrain_float((c->get_output_norm() + 1.0) * 0.5, 0, 1) * 65535);
```

Comment:
This method uses the servos functions individual endpoints. `SERVO1_MIN` ect. However if its a motor its actually using the motors endpoints `MOT_PWM_MIN`. 

The other drivers typically calculate there own normalized output using a hard coded range of 1000 to 2000 pwm. 

However that is only true for motors, I guess here the end goal would be to pass both a "motor" command and a "servo" command?

Maybe a bit mask for each?